code/modules/mob/inventory.dm

```diff
@@ -236,6 +245,8 @@ var/list/slot_equipment_priority = list( \
 /mob/proc/unEquip(obj/item/I, var/atom/target)
 	if(!canUnEquip(I))
 		return
+	if(target && !is_space_available(I, get_turf(target)))
```

Comment:
This proc moves `I` into `target`, not onto `target`'s turf.

Reply:
As designed.

Reply:
The check is at the turf level to prevent bypassing content restrictions by nesting contents in contents /w containers.